tendermint/src/public_key.rs

```diff
@@ -1,73 +1,73 @@
 //! Public keys used in Tendermint networks
 
+pub use ed25519_dalek::PublicKey as Ed25519PublicKey;
+
+#[cfg(feature = "secp256k1")]
+pub use k256::PublicKey as Secp256k1PublicKey;
+
 use crate::error::{Error, Kind};
 use anomaly::fail;
-use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
-#[cfg(feature = "secp256k1")]
-use signatory::ecdsa::curve::secp256k1;
-use signatory::ed25519;
-use std::{
-    fmt::{self, Display},
-    ops::Deref,
-    str::FromStr,
-};
-use subtle_encoding::base64;
-use subtle_encoding::{bech32, hex};
+use serde::{de, ser, Deserialize, Serialize};
+use std::{cmp::Ordering, fmt, ops::Deref, str::FromStr};
+use subtle_encoding::{base64, bech32, hex};
 
 /// Public keys allowed in Tendermint protocols
-#[derive(Serialize, Deserialize)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(tag = "type", content = "value")]
-#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
 pub enum PublicKey {
     /// Ed25519 keys
     #[serde(
         rename = "tendermint/PubKeyEd25519",
         serialize_with = "serialize_ed25519_base64",
         deserialize_with = "deserialize_ed25519_base64"
     )]
-    Ed25519(ed25519::PublicKey),
+    Ed25519(Ed25519PublicKey),
 
     /// Secp256k1 keys
     #[cfg(feature = "secp256k1")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "secp256k1")))]
     #[serde(
         rename = "tendermint/PubKeySecp256k1",
         serialize_with = "serialize_secp256k1_base64",
         deserialize_with = "deserialize_secp256k1_base64"
     )]
-    Secp256k1(secp256k1::PublicKey),
+    Secp256k1(Secp256k1PublicKey),
```

Comment:
If you want to keep this gated under a `secp256k1` feature, it might make sense to make `pub enum PublicKey` a `#[non_exhaustive]` enum. Otherwise clients might assume that variants do or do not exist depending on what features are enabled, and if they try to exhaustively match, they'll break if things don't match up to their expectations.

Or, perhaps the (lightweight, pure Rust) `k256` crate is a less onerous dependency than `signatory-secp256k1` (C library wrapper) to the point that the `secp256k1` feature can be removed.

Reply:
All for the `enum` with non_exhaustive - how much would it take to remove the `secp256k1` feature?

Reply:
Remove it as a feature, or remove all secp256k1-related code?

For what it's worth, the KMS still relies on it for the transaction signing feature.